### PR TITLE
[CI] Add inductor workflow for rocm

### PR DESCRIPTION
**Pull request by pragupta**
This PR is to create a separate CI job for inductor UTs on ROCm. You will need to add `ciflow/inductor` tag on PRs to trigger this job. However, the job will run on its own on any commit merged in main. This job takes around 1.5 hours to run and it is run in parallel to other rocm jobs. It is run only on the MI210 CI runners to ensure maximum inductor functionality is tested.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
Waiting on https://github.com/pytorch/pytorch/pull/110511 to be merged before enabling this workflow so as to not overburden ROCm CI.

**Comment by pragupta** [comment]
Saving the passing CI link before rebasing: https://hud.pytorch.org/pr/110544

**Comment by jithunnair-amd** [comment]
> Saving the passing CI link before rebasing: https://hud.pytorch.org/pr/110544

Be aware that the link is dynamic; it will also update when you rebase :)

**Comment by huydhn** [comment]
Please help double check the duration of the new test job too.  I'm waiting for the current run to finish https://github.com/pytorch/pytorch/actions/runs/6894570049/job/18757398102 to see how long it takes to finish.  More shards might be needed if it takes more than 2 hours

File: .ci/pytorch/test.sh [hunk 1]
```diff
@@ -18,6 +18,10 @@ BUILD_DIR="build"
 BUILD_RENAMED_DIR="build_renamed"
 BUILD_BIN_DIR="$BUILD_DIR"/bin
 
+#Set Default values for these variables in case they are not set
+SHARD_NUMBER="${SHARD_NUMBER:=1}"
+NUM_TEST_SHARDS="${NUM_TEST_SHARDS:=1}"
```
Comment (malfet) [inline .ci/pytorch/test.sh:23]: Why this is needed? I would prefer it to error out if SHARD_NUMBER is not defined

**Comment by pragupta** [comment]
> Please help double check the duration of the new test job too. I'm waiting for the current run to finish https://github.com/pytorch/pytorch/actions/runs/6894570049/job/18757398102 to see how long it takes to finish. More shards might be needed if it takes more than 2 hours

@huydhn -- I have noticed that it takes around 1.5 hours on average for this job to run. Is that sufficient for one shard?

File: .ci/pytorch/test.sh (hunk 1 above)
Comment (pragupta) [reply .ci/pytorch/test.sh:23]: We added these to simplify our internal testing.

**Comment by huydhn** [comment]
> @huydhn -- I have noticed that it takes around 1.5 hours on average for this job to run. Is that sufficient for one shard?

I noticed that yesterday. 1.5h is acceptable, but the ROCm inductorc job failed and terminated early, so let's wait once it finishes successfully and get the number then

**Review by jithunnair-amd — APPROVED** [review]

File: .ci/pytorch/test.sh (hunk 1 above)
Comment (pragupta) [reply .ci/pytorch/test.sh:23]: In order to run inductor config, it seems that SHARD_NUMBER is required now. (https://github.com/pytorch/pytorch/pull/110544/files#diff-9709e5db13aeac90c0312b2b8da34b37cf51242ec789ca959fcf1ba295a8da7aR1101)

Although this variable is always defined in all the CI  yaml files that trigger these tests, it's nice to be able to run this file locally outside of CI context.

**Comment by jithunnair-amd** [comment]
@huydhn @malfet Can you please approve and merge this PR?

**Review by jansel — APPROVED** [review]

**Review by huydhn — APPROVED** [review]

**Pull request closed without merging**